client/components/RecipeDetails/RecipeDetails.jsx

```diff
@@ -35,128 +35,122 @@ registerPlugin(
   FilePondPluginImageResize
 );
 
-class RecipeDetails extends Component {
-  static displayName = 'RecipeDetails';
-  static propTypes = {
-    className: PropTypes.string,
-    recipe: PropTypes.object,
-    recipeMods: PropTypes.arrayOf(PropTypes.object),
-    saveAlteration: PropTypes.func,
-    addPhoto: PropTypes.func,
-    photosLength: PropTypes.number,
-    client: PropTypes.instanceOf(ApolloClient)
-  };
-
-  state = {
-    errors: {},
-    edits: {},
-    timeouts: {},
-    editing: !this.props.recipe ? true : false
+const RecipeDetails = ({
+  recipe,
+  addPhoto,
+  className,
+  photosLength,
+  recipeMods,
+  saveAlteration,
+  client
+}) => {
+  const [errors, setErrors] = useState({});
+  const [edits, setEdit] = useState({});
+  const [editing, setEditing] = useState(!recipe ? true : false);
+  const [timeouts, setTimeoutValue] = useState({});
+
+  const containerRef = useRef(null);
+  const titleInputRef = useRef(null);
+  const descriptionInputRef = useRef(null);
+  const timeInputRef = useRef(null);
+  const servingInputRef = useRef(null);
+  let pond;
+
+  useEffect(() => {
+    return () => {
+      document.removeEventListener('mousedown', handleClick);
+    };
+  }, []);
+
+  const enableEditing = async refTitle => {
+    if (refTitle) {
+      edit(refTitle);
+    } else {
+      setEditing(true);
+      document.addEventListener('mousedown', handleClick);
+    }
```

Comment:
You can move the logic from the "edit" function into this function because enableEditing always needs to 3 things:
1. call `await setEditing(true)`
2. if a fieldName was supplied, focus on that field
3. add the mousedown handler

Reply:
is it fine now ?

Reply:
Nice! this is much easier to read. I still think you could move this whole block of logic into the enableEditing function since that is the only place "edit" ever gets called. That would also make the title input focus when no fieldname is provided to enableEditing, which would be a good default behavior.

Reply:
I have moved all the logic from edit function to enableEdit function.

Reply:
Excellent!